docs/concepts/components/callbacks.md

```diff
@@ -2,27 +2,92 @@
+
+If the given vector is empty, this function doesn't do anything.
+
 ### callback
 
-Registers a callback that will send a message to the component's update mechanism when it is executed. Under the hood, it will call `send_self` with the message that is returned by the provided closure. A `Fn(IN) -> COMP::Message` is provided and a `Callback<IN>` is returned.
+Create a callback that will send a message to the component when it is executed.
+Under the hood, it will call `send_message` with the message that is returned by the provided closure.
+
+This method has a version that accepts an `FnOnce` instead, `callback_once`.
+You should use this with care though, as the resulting callback will panic if executed more than once.
+
+```rust
+// Create a callback that accepts some text and sends it to the component as the `Msg::Text` message variant.
+let cb = link.callback(|text: String| Msg::Text(text));
+
+// The previous line is needlessly verbose to make it clearer.
+// It can be simplified it to this:
+let cb = link.callback(Msg::Text);
 
-### send\_message
+// Will send `Msg::Text("Hello World!")` to the component.
+cb.emit("Hello World!".to_owned());
+```
 
-Sends a message to the component immediately after the current loop finishes, causing another update loop to initiate.
+### batch_callback
 
-### batch\_callback
+Create a callback that will send a batch of messages to the component when it is executed.
+The difference to `callback` is that the closure passed to this method doesn't have to return a message.
+Instead, the closure can return either `Vec<Msg>` or `Option<Msg>` where `Msg` is the component's message type.
 
-Registers a callback that sends a batch of many messages at once when it is executed. If any of the messages cause the component to re-render, the component will re-render after all messages in the batch have been processed. A `Fn(IN) -> Vec<COMP::Message>` is provided and a `Callback<IN>` is returned.
+`Vec<Msg>` is treated as a batch of messages and uses `send_message_batch` under the hood.
+
+`Option<Msg>` calls `send_message` if it is `Some`. If the value is `None`, nothing happens.
+This can be used in cases where, depending on the situation, an update isn't required.
+
+This is achieved using the `SendAsMessage` trait which is only implemented for these types.
+You can implement `SendAsMessage` for your own types which allows you to use them in `batch_callback`.
+
+Like `callback`, this method also has a `FnOnce` counterpart, `batch_callback_once`.
+The same restrictions apply as for `callback_once`.
 
 ## Callbacks
 
 _\(This might need its own short page.\)_
 
-Callbacks are used to communicate with services, agents, and parent components within Yew. They are just a `Fn` wrapped by an `Rc` to allow them to be cloned.
+Callbacks are used to communicate with services, agents, and parent components within Yew. They are just an `Fn` wrapped by an `Rc` to allow them to be cloned.
```

Comment:
I'm not sure if this is the correct form of the article for `Fn` (`Fn` is short "function" which would be "a function" rather than "an function"). 🤷🏼 
```suggestion
Callbacks are used to communicate with services, agents, and parent components within Yew. They are just an `Fn` wrapped by an `Rc` to allow them to be cloned.
```

Reply:
It's always kind of jarring in these situations. "Fn" might be short for function but when I see it like that I read it as "eff-enn" (not exactly IPA but it will have to do :D) which most definitely starts with a vowel sound.
Perhaps we should try to avoid adding articles to types?

```suggestion
Callbacks are used to communicate with services, agents, and parent components within Yew.
Internally their type is just `Fn` wrapped in `Rc` to allow them to be cloned.
```

Reply:
Yeah sounds good. I'm not particularly fussed about the rules we have so long as we apply the rules we set consistently. 😁